File: DotNetGotchas/VBDN/CopyingObjects/Modified3/Copy/Brain.vb
'Brain.vb
Public Class Brain
    Implements ICloneable

    Public Sub New()

    End Sub

    Public Sub New(ByVal another As Brain)
        ' Code to properly copy Brain can go here
    End Sub

    Public Overrides Function ToString() As String
        Return Me.GetType().Name & ":" & GetHashCode()
    End Function

    '...
    Public Function Clone() As Object _
        Implements System.ICloneable.Clone
        Return MemberwiseClone()
    End Function
End Class

File: DotNetGotchas/CSharp/CopyingObjects/Modified3/Copy/Brain.cs
//Brain.cs
using System;

namespace Copy
{
	public class Brain : ICloneable
	{
		public Brain() {}

		public Brain(Brain another) 
		{
			//Code to properly copy Brain can go here
		}

		public override string ToString()
		{
			return GetType().Name + ":" + GetHashCode();
		}

		//...
		#region ICloneable Members

		public object Clone()
		{
			return MemberwiseClone();
		}

		#endregion
	}
}
